terraform/dev/main.tf

```diff
@@ -66,3 +69,35 @@ resource "cloudfoundry_service_instance" "document_upload_bucket" {
   space        = data.cloudfoundry_space.space.id
   service_plan = data.cloudfoundry_service.s3.service_plans["basic"]
 }
+
+###
+# ClamAV networking
+###
+
+data "cloudfoundry_app" "clamav_server" {
+  name_or_id = var.clamav_server_app_name
+  space      = data.cloudfoundry_space.space.id
+}
+
+data "cloudfoundry_app" "clamav_rest" {
+  name_or_id = var.clamav_rest_app_name
+  space      = data.cloudfoundry_space.space.id
+}
+
+data "cloudfoundry_app" "ttahub" {
+  name_or_id = "tta-smarthub-${var.env}"
+  space      = data.cloudfoundry_space.space.id
+}
```

Comment:
The obvious problem here is that the three preceding steps fail if the apps don't exist yet. Not really a problem for the two clam apps, since they can be deployed first without error, and for dev it's not a problem because the app and db services already exist.

For production, there's a chicken and egg problem now where `terraform apply` won't cleanly work unless the prod app exists, but the prod app depends on the db services. Any ideas on that?

Reply:
After conversation with @dcmcand we decided that commenting out this section for the initial bootstrap run is a fine way to break the circular dependency loop.

There may be "cleaner" ways to do it, but the cost/payoff of commenting out is worth the most to us.